react/base.yml

```diff
@@ -0,0 +1,57 @@
+# Enforce best practices and protect against common mistakes in JSX and React.
+#
+# This is an alternative to openstax-poland/base which includes additional rules
+# from eslint-plugin-react.
+
+extends:
+  - ../base.yml
+  - plugin:react/recommended
+
+settings:
+  react:
+    version: 16.2.0
+
+rules:
+  # Forbid use of .bind() and arrow functions as JSX callbacks, as they are not
+  # comparable and can cause unnecessary re-renders.
+  react/jsx-no-bind: error
```

Comment:
I would make this `warn` since it will cause multiple errors at start when we will applies those rules and then we can change it one by one when updating components.

Reply:
Are we actually using bind or arrow functions anywhere? I thought I was rejecting any PRs with them.

(Note that this refers to `.bind()` and arrow functions _inside_ the render function, as this will create a new function each time `render` is called. It's still fine to use them outside of `render`)

Reply:
We have 38 arrow functions inside `render` since sometimes it's just easier to pass `() => this.removeAlert(alert)` than assigning `alert.id` as `data-id` and then searching alert by this `id` in array and then calling `removeAlert`.

However I agree that it should be done. I'm just not sure if I want to spent few hours straight on doing this :D

Reply:
After reconsider I think I will do this as one task for implementing eslint and tslint to `adaptarr-front`